libindy/src/domain/crypto/did.rs

```diff
@@ -1,43 +1,144 @@
 use named_type::NamedType;
 
+use regex::Regex;
+use rust_base58::FromBase58;
+
+use utils::validation::Validatable;
+use utils::qualifier;
+
+pub const ABBREVIATABLE_PREFIX: &'static str = "did:sov";
+
 #[derive(Serialize, Deserialize, Clone, Debug)]
 pub struct MyDidInfo {
-    pub did: Option<String>,
+    pub did: Option<DidValue>,
     pub seed: Option<String>,
     pub crypto_type: Option<String>,
-    pub cid: Option<bool>
+    pub cid: Option<bool>,
+    pub method_name: Option<String>,
+}
+
+impl Validatable for MyDidInfo {
+    fn validate(&self) -> Result<(), String> {
+        if let Some(ref did) = self.did {
```

Comment:
Could you use `did_doc` from crates.io?

Reply:
libindy `MyDidInfo` is significantly diverse from W3C DIDDoc